pysynchronizer/objects.py

```diff
@@ -0,0 +1,188 @@
+                self.connected = True
+            except Exception as err:
+               raise ConnectionError('Failed to connect to XenMgr service') from err
+        else:
+            self.uuid=""
+            self.connected = False
+
+    def delete(self):
+        """Attempts to delete the VM ref, returns True if successful"""
+        if self.connected:
+            self.proxy_object.delete()
+            self.uuid = None
+            self.proxy_object = None
+            self.connected = False
+            return True
+
+        return False
+
+    def get_property(self, name):
+        """Retrieve a DBus object property, returning "" for when it does not exist"""
+        try:
+            value = self.proxy_object.get_property(name)
+            return str(value)
+        except:
+            return ""
+
+    def set_property(self, name, value):
+        """Set the value of a DBus object property, returning value if successful"""
+        try:
+            self.proxy_object.set_property(name, value)
+            return value
+        except:
+            return ""
+
+    def domstore_get(self, key):
+        """Retrieve a Domstore key, returning "" for when it does not exist"""
+        try:
+            value = self.proxy_object.get_domstore_key(key)
+            return str(value)
+        except:
+            return ""
+
+    def domstore_set(self, key, value):
+        """Set the value of a Domstore key, returning value if successful"""
+        try:
+            self.proxy_object.set_domstore_key(key, value)
+            return value
+        except:
+            return ""
+
+    def disks(self):
+        """Retrieve a list of VMDisk objects of the disk associate with the VM"""
+        disks = []
+        for disk_path in self.proxy_object.list_disks():
+            disks.append(VMDisk(disk_path))
+
+        return disks
+
+    def find_disk(self, dsk_id):
+        """Search if VM has an associated disk matching the disk identifier"""
+        for disk in self.disks():
+            values = [
+                disk.id,
+                str(disk.get_property("phys-path")),
+                str(disk.get_property("virt-path")),
+            ]
+            if dsk_id in values:
+                return disk
+
+        return None
+
+class VMDisk:
+    def __init__(self, path):
+            self.id = path.rsplit('/',1)[1]
+            self.proxy_object = OXTDBusApi.open_disk(path)
+
+    def get_property(self, name):
+        """Retrieve a DBus object property, returning "" for when it does not exist"""
+        return self.proxy_object.get_property(name)
+
+    def set_property(self, name, value):
+        """Set the value of a DBus object property, returning value if successful"""
+        return self.proxy_object.set_property(name, value)
+
+    def name(self):
+        """Returns the phys-path as the "name" of the disk"""
+        path = self.get_property('phys-path')
+        return path.rsplit('/',1)[1]
```

Comment:
I had to look up rsplit, so...  This is probably fine for this use, but it is a little fragile:
```
>>> "hdc".rsplit('/',1)[1]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
```
If you just do `path.split('/')[-1]` you get the last element without the fragility.

Reply:
Actually that doesn't avoid it, the problem is that either way they are assumptive and will fail. The rsplit fails with an exception at the point of indexing the result and split will succeed but will result in an empty string which will cause a different failure later when it is attempted to be used. The answer is that either is fine, for rsplit I will need to check that the result is of len 2 or that the value of split('/')[-1] is not of len 0.